src/tools/clippy/clippy_utils/src/check_proc_macro.rs

```diff
@@ -149,7 +149,7 @@ fn expr_search_pat(tcx: TyCtxt<'_>, e: &Expr<'_>) -> (Pat, Pat) {
             (Pat::Str("for"), Pat::Str("}"))
         },
         ExprKind::Match(_, _, MatchSource::Normal) => (Pat::Str("match"), Pat::Str("}")),
-        ExprKind::Match(e, _, MatchSource::TryDesugar) => (expr_search_pat(tcx, e).0, Pat::Str("?")),
+        ExprKind::Match(e, _, MatchSource::TryDesugar(_)) => (expr_search_pat(tcx, e).0, Pat::Str("?")),
```

Comment:
This might be the reason for the Clippy regression. Did the expression `e` that is stored in the `ExprKind::Match` change by any chance and that's why re-calling `expr_search_pat` leads to different results than before? :thinking: 

(Pretty wild guess)

Reply:
Fixing in https://github.com/rust-lang/rust/pull/115122